From 86b54e493d657e9aaa2416ee9fec4924052353da Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Tue, 25 Nov 2025 21:42:20 +0100
Subject: [PATCH] Unzip XML when opening file

Without this, the recent versions of libxml does not try to unzip the file
content, causes a read failure.
---
 src/xml-label.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/xml-label.c b/src/xml-label.c
index 293c0745..725b1150 100644
--- a/src/xml-label.c
+++ b/src/xml-label.c
@@ -195,7 +195,7 @@ gl_xml_label_open (const gchar      *utf8_filename,
 	filename = g_filename_from_utf8 (utf8_filename, -1, NULL, NULL, NULL);
 	g_return_val_if_fail (filename, NULL);
 
-        doc = xmlReadFile (filename, NULL, XML_PARSE_HUGE);
+        doc = xmlReadFile (filename, NULL, XML_PARSE_HUGE | XML_PARSE_UNZIP);
 	if (!doc) {
 		g_message ("xmlParseFile error");
 		*status = XML_LABEL_ERROR_OPEN_PARSE;
